src/main/java/teammates/common/util/EmailWrapper.java

```diff
@@ -0,0 +1,89 @@
+package teammates.common.util;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents an email message and its important metadata.
+ */
+public class EmailWrapper {
```

Comment:
I'd like to name this class `Email`, but it's the same name used by Mailgun and Sendgrid's current implementation.